src/modm/math/geometry/vector.hpp

```diff
@@ -11,141 +12,552 @@
-	 *
-	 * \section	point_vector	Point vs. vector
+	 * @param args		scalars, Vectors and Arrays in your desired order
+	 */
+	template <class... Args>
+	constexpr Vector(Args... args)
+	{ assign<0>(args...); }
+
+	template <typename U>
+	requires std::convertible_to<U, T>
+	constexpr explicit Vector(U v)
+	{
+		if constexpr(std::integral<T> && std::floating_point<U>)
+			this->fill(std::round(v));
+		else
+			this->fill(v);
+	}
+
+	template <typename U>
+	constexpr Vector(const U (&arr)[N])
+	{
+		if constexpr(std::integral<T> && std::floating_point<U>)
+			std::transform(std::begin(arr), std::end(arr), this->begin(), [] (U v) { return std::round(v); });
+		else
+			std::copy(std::begin(arr), std::end(arr), this->begin());
+	}
+
+	// TODO matrix constructor
+	/* constexpr Vector(const Matrix<T, N, 1> &rhs)
+	{ std::copy(std::begin(rhs.element), std::end(rhs.element), this->begin()); } */
+
+	// matrix assignment
+	/* constexpr Vector& operator= (const Matrix<T, N, 1> &rhs)
+	{
+		std::copy(std::begin(rhs.element), std::end(rhs.element), this->begin());
+		return *this;
+	} */
+
+	// accessors for x, y, z, w, v, u
+	constexpr T x() const { return this->operator[](0); }
+	constexpr T& x() { return this->operator[](0); }
+
+	constexpr T y() const requires (N > 1) { return this->operator[](1); }
+	constexpr T& y() requires (N > 1) { return this->operator[](1); }
+
+	constexpr T z() const requires (N > 2) { return this->operator[](2); }
+	constexpr T& z() requires (N > 2) { return this->operator[](2); }
+
+	constexpr T w() const requires (N > 3) { return this->operator[](3); }
+	constexpr T& w() requires (N > 3) { return this->operator[](3); }
+
+	constexpr T v() const requires (N > 4) { return this->operator[](4); }
+	constexpr T& v() requires (N > 4) { return this->operator[](4); }
+
+	constexpr T u() const requires (N > 5) { return this->operator[](5); }
+	constexpr T& u() requires (N > 5) { return this->operator[](5); }
+
+private:
+	/**
+	 * @brief 			Creates a new Vector by applying a binary function between each
+	 * 					item of this and each item of other
 	 *
-	 * In geometry, it is often convenient to use vector arithmetic to
-	 * represent points.
+	 * @tparam 	Func 	Binary function
+	 */
+	template <class BinFunc, typename U = T>
+	constexpr Vector<U, N>
+	calcNewVector(const Vector &other) const
+	{
+		Vector<U, N> res;
+
+		// no clue why, but RVO / Loop unrolling is not applied for this ...
+		// std::transform(this->begin(), this->end(), other.begin(), res.begin(), Func{});
+
+		// ... but this optimizes perfectly.
+		res.x() = BinFunc{}(this->x(), other.x());
+		if constexpr(N > 1) res.y() = BinFunc{}(this->y(), other.y());
+		if constexpr(N > 2) res.z() = BinFunc{}(this->z(), other.z());
+		if constexpr(N > 3) res.w() = BinFunc{}(this->w(), other.w());
+		if constexpr(N > 4) res.v() = BinFunc{}(this->v(), other.v());
+		if constexpr(N > 5) res.u() = BinFunc{}(this->u(), other.u());
+
```

Comment:
For `operator`s returning a new `Vector`, the `<algorithm>` was not optimize suficiently - execution time has tripled. I've found a workaround, using these `if constexpr` constructs but there's for sure a more elegant solution.